### PR TITLE
*: make session variables control TiKV client variables.

### PR DESCRIPTION
**Pull request by coocood**
And add a system variable `tidb_backoff_lock_fast` to change the backoff base time on read lock.

### CODE REVIEW

**Comment by coocood** [comment]
@tiancaiamao @jackysp @zz-jason PTAL

**Comment by shenli** [comment]
What's the benefit of this PR?

File: distsql/distsql.go [hunk 1]
```diff
@@ -223,7 +223,7 @@ func Select(ctx context.Context, sctx sessionctx.Context, kvReq *kv.Request, fie
 	if !sctx.GetSessionVars().EnableStreaming {
 		kvReq.Streaming = false
 	}
-	resp := sctx.GetClient().Send(ctx, kvReq)
+	resp := sctx.GetClient().Send(ctx, kvReq, sctx.GetSessionVars().KVVars)
```
Comment (tiancaiamao) [inline distsql/distsql.go]: Why not put it in kvReq ?

Comment (coocood) [reply distsql/distsql.go]: the variables don't belong to the request.
It's long-term variables, only changed by explicit `SET` statement.

**Comment by tiancaiamao** [comment]
LGTM

**Comment by jackysp** [comment]
I am testing the effect of performance of this pr. So wait to merge.

**Pull request merged**